src/osd/PGLog.cc

```diff
@@ -674,16 +674,20 @@ void PGLog::_write_log_and_missing_wo_missing(
   set<string> *log_keys_debug
   )
 {
-  set<string> to_remove(trimmed_dups);
-  for (set<eversion_t>::const_iterator i = trimmed.begin();
-       i != trimmed.end();
+  set<string> to_remove;
+  to_remove.swap(*trimmed_dups);
```

Comment:
dereferencing a `nullptr`, will crash the application. and we always pass `nullptr`s to this function.
see https://github.com/ceph/ceph/pull/19099/commits/c2e2f03ec51c7ed3ea6b56f5a16f59ed0db60aa8#diff-2bf6ec52b400c95e6a23a186482477b6R621